daphne/src/vdaf/mod.rs

```diff
@@ -75,12 +77,142 @@ impl AsRef<[u8]> for VdafVerifyKey {
     }
 }
 
-/// An Aggregator's inputs for VDAF preparation.
+/// Report state during aggregation initialization after consuming the report share. This involves
+/// decryption as well a few validation steps.
 #[derive(Clone, Debug, PartialEq)]
-pub struct VdafPrepInput<'req> {
-    pub(crate) public_share: Cow<'req, Vec<u8>>,
-    pub(crate) input_share: Vec<u8>,
-    pub(crate) metadata: Cow<'req, ReportMetadata>,
+pub enum EarlyReportStateConsumed<'req> {
+    Ready {
+        metadata: Cow<'req, ReportMetadata>,
+        public_share: Cow<'req, [u8]>,
+        input_share: Vec<u8>,
+    },
```

Comment:
The use of `Cow` here seems unnecessary, unless I'm missing something

Reply:
So my thinking is that we're going to want an owned variant of this struct when we plumb it in DO. I'll remove this if it turns out not to be necessary.